materialize-databricks/driver.go

```diff
@@ -0,0 +1,682 @@
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"text/template"
+	"time"
+
+	_ "github.com/databricks/databricks-sql-go"
+	dbsqlerr "github.com/databricks/databricks-sql-go/errors"
+	"github.com/databricks/databricks-sdk-go"
+	dbConfig "github.com/databricks/databricks-sdk-go/config"
+	driverctx "github.com/databricks/databricks-sql-go/driverctx"
+	boilerplate "github.com/estuary/connectors/materialize-boilerplate"
+	sql "github.com/estuary/connectors/materialize-sql"
+	pf "github.com/estuary/flow/go/protocols/flow"
+	pm "github.com/estuary/flow/go/protocols/materialize"
+	log "github.com/sirupsen/logrus"
+	"go.gazette.dev/core/consumer/protocol"
+	"golang.org/x/sync/errgroup"
+)
+
+const defaultPort = "443"
+const volumeName = "flow_staging"
+
+type tableConfig struct {
+	Table         string `json:"table" jsonschema:"title=Table,description=Name of the table" jsonschema_extras:"x-collection-name=true"`
+	Schema        string `json:"schema" jsonschema:"title=Schema,description=Schema where the table resides,default=default"`
+	Delta         bool   `json:"delta_updates,omitempty" jsonschema:"default=false,title=Delta Update,description=Should updates to this table be done via delta updates. Default is false."`
+}
+
+func newTableConfig(ep *sql.Endpoint) sql.Resource {
+	cfg := ep.Config.(*config)
+  var schema = cfg.SchemaName
+  if schema == "" {
+    schema = "default"
+  }
+	return &tableConfig{Schema: schema}
+}
+
+// Validate the resource configuration.
+func (r tableConfig) Validate() error {
+	var forbiddenChars = ". /"
+	if r.Table == "" {
+		return fmt.Errorf("missing table")
+	}
+	if strings.ContainsAny(r.Table, forbiddenChars) {
+		return fmt.Errorf("table name %q contains one of the forbidden characters %q", r.Table, forbiddenChars)
+	}
+
+	if r.Schema == "" {
+		return fmt.Errorf("missing schema")
+	}
+	if strings.ContainsAny(r.Schema, forbiddenChars) {
+		return fmt.Errorf("schema name %q contains one of the forbidden characters %q", r.Schema, forbiddenChars)
+	}
+
+	return nil
+}
+
+func (c tableConfig) Path() sql.TablePath {
+	return []string{c.Schema, c.Table}
+}
+
+func (c tableConfig) GetAdditionalSql() string {
+	return ""
+}
+
+func (c tableConfig) DeltaUpdates() bool {
+	return c.Delta
+}
+
+func newDatabricksDriver() *sql.Driver {
+	return &sql.Driver{
+		DocumentationURL: "https://go.estuary.dev/materialize-databricks",
+		EndpointSpecType: new(config),
+		ResourceSpecType: new(tableConfig),
+		NewEndpoint: func(ctx context.Context, raw json.RawMessage, tenant string) (*sql.Endpoint, error) {
+			var cfg = new(config)
+			if err := pf.UnmarshalStrict(raw, cfg); err != nil {
+				return nil, fmt.Errorf("parsing endpoint configuration: %w", err)
+			}
+
+			log.WithFields(log.Fields{
+				"address":  cfg.Address,
+				"path": cfg.HTTPPath,
+        "catalog": cfg.CatalogName,
+			}).Info("connecting to databricks")
+
+			var metaBase sql.TablePath
+			var metaSpecs, _ = sql.MetaTables(metaBase)
```

Comment:
Does the creation of meta tables respect the `schema_name` configured in the endpoint config? I might be missing how that is threaded through. I'm wondering if you set a `schema_name` in the endpoint config to something other than `default`, if the metadata tables get created in that alternate schema.